tests/instrumentation-framework/src/main/java/org/hawkular/apm/tests/dockerized/environment/DockerImageExecutor.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.tests.dockerized.environment;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hawkular.apm.tests.dockerized.InterfaceIpV4Address;
+import org.hawkular.apm.tests.dockerized.exception.EnvironmentException;
+import org.hawkular.apm.tests.dockerized.model.TestEnvironment;
+import org.hawkular.apm.tests.dockerized.model.Type;
+
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.LogStream;
+import com.spotify.docker.client.exceptions.DockerCertificateException;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.HostConfig;
+
+/**
+ * @author Pavol Loffay
+ */
+public class DockerImageExecutor implements TestEnvironmentExecutor {
+    private static final String APM_AGENT_DIRECTORY = "/opt/hawkular-apm-agent";
```

Comment:
BTW this is directory inside docker container, which is mounted from host OS to access agent. So it can be anything.